File: 단계별로 풀어보기/함수/1065/한수.cpp
#include <iostream>
using namespace std;

int findHanSu(int N){
    int cnt=0;
    bool same = true;
    
    for(int i=0; i<=N; i++){
        int differ = (j%100-j%10)/10-(j%10);
        
        if(differ != (j%1000-j%100)/100-(j%100-j%10)/10)
            continue;
            
            
    }

    
    return cnt;
}

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int N;
    cin>>N;
    
    return 0;
}

File: 단계별로 풀어보기/함수/1065/한수.py
import sys

def countHansu(N):
    cnt = 0
    
    for i in range(1,N+1):
        same = True
        j = i
        while(same):
            if j//100 == 0:
                break
            differ = ((j%1000)-(j%100))//100-((j%100)-(j%10))//10
            if(differ != ((j%100)-(j%10))//10-(j%10)):
                same = False
                break
            j=j//10
            
        if same == True:
            cnt+=1
    
    return cnt

def init():
    N=int(sys.stdin.readline())
    print(countHansu(N))

init()
